core/lib/zksync_core/src/metadata_calculator/updater.rs

```diff
@@ -174,6 +174,8 @@ impl TreeUpdater {
             // That is, if we run multiple tree instances, we'll get metadata correspondence
             // right away without having to implement dedicated code.
 
+            // If there is an `object_key`, then the witness is being processed.
+            // If not, it must've been skipped (witness/proving is done at a very small percentage on testnet).
```

Comment:
Nit: I don't quite understand this comment. `object_key` is obtained from `process_l1_batch()` above, and it's `Some` for full trees and `None` for lightweight ones. How does prover sampling factor into this?